File: app/src/main/java/com/example/miodragmilosevic/nativetest/filters/javaFilters/SnowFilter.java
package com.example.miodragmilosevic.nativetest.filters.javaFilters;

import android.graphics.Color;

import com.example.miodragmilosevic.nativetest.filters.ImageFilter;

import java.util.Random;

/**
 * Created by miodrag.milosevic on 12/15/2017.
 */

public class SnowFilter extends ImageFilter {
    private static final int COLOR_MAX = 255;

    @Override
    protected int[] transformPixels(int[] pixelArray, int imageWidth, int imageHeight) {
        Random random = new Random();

        int  threshHold = 0;
        // iteration through pixels
        // Get original pixels

        int A, R, G, B;
        for (int i = 0; i < pixelArray.length; i++) {
            A = Color.alpha(pixelArray[i]);
            R = Color.red(pixelArray[i]) ;
            G = Color.green(pixelArray[i] );
            B = Color.blue(pixelArray[i]) ;
            threshHold = random.nextInt(COLOR_MAX);

            if(R > threshHold && G > threshHold && B > threshHold) {
                pixelArray[i] = Color.argb(A,COLOR_MAX, COLOR_MAX, COLOR_MAX);
            } else {
                pixelArray[i] = Color.argb(A, R, G, B);
            }
        }
        return pixelArray;
    }
}

File: app/src/main/cpp/SnowFilter.cpp
//
// Created by miodrag.milosevic on 12/29/2017.
//

#include <stdlib.h>
#include "SnowFilter.h"
#include "color.h"


void SnowFilter::transformPixels(jint *pixelArray, jint imageWidth, jint imageHeight) {
    jint  threshHold = 0;
// iteration through pixels
// Get original pixels

    jint A, R, G, B;
    for (jint i = 0; i < imageWidth*imageHeight; i++) {
        A = Color::alpha(pixelArray[i]);
        R = Color::red(pixelArray[i]) ;
        G = Color::green(pixelArray[i] );
        B = Color::blue(pixelArray[i]) ;
        threshHold = rand()%COLOR_MAX;

        if(R > threshHold && G > threshHold && B > threshHold) {
            pixelArray[i] = Color::argb(A,COLOR_MAX, COLOR_MAX, COLOR_MAX);
        } else {
            pixelArray[i] = Color::argb(A, R, G, B);
        }
    }
}
